bfcfg

```diff
@@ -920,15 +923,83 @@ sys_redfish_cfg()
   rm -f ${tmp_file}
 }
 
+#
+# Dump the boot watchdog configuration.
+# Below are the offsets defined in UEFI which is 4(fixed header) plus the offset
+# of the variable within the BfBootWdog struct. These offsets are not supposed to
+# change in order to be backward compatible with previous releases.
+#   VARIABLE(Name)       OFFSET(Byte)  SIZE(Byte)
+#   INTERVAL                  4            2
+#   MODE                      5            1
+#
+sysfs_dump_boot_wdog()
+{
+  local current_mode current_interval current_mode_str
+  local next_mode next_interval next_mode_str
+  local tmp_file=${TMP_DIR}/.bfcfg-bootwdog-data
+
+  if [ ${dump_level} -eq ${DUMP_LEVEL_NONE} ]; then
+    return
+  fi
+
+  if [ ! -e "${boot_wdog_sysfs}" ]; then
+    log_msg "misc: failed to find the ${boot_wdog_sysfs} EFI variable"
+    return
+  fi
+
+  # shellcheck disable=SC2216
+  yes | cp -f ${boot_wdog_sysfs} ${tmp_file}
+  current_interval=$(hexdump -s 4 -n 2 -e '"%d" "\n"' "${tmp_file}")
+  current_mode=$(hexdump -s 5 -n 1 -e '"%d" "\n"' "${tmp_file}")
+
+  if [ "${current_mode}" -eq 0 ]; then
+    current_mode_str="DISABLED"
+  elif [ "${current_mode}" -eq 1 ]; then
+    current_mode_str="STANDARD"
+  elif [ "${current_mode}" -eq 2 ]; then
+    current_mode_str="TIME_LIMIT"
+  else
+    current_mode_str="UNKNOWN"
+  fi
+
+  echo "misc: CURRENT_BOOT_WDOG_MODE=${current_mode_str}"
+  echo "misc: CURRENT_BOOT_WDOG_INTERVAL=${current_interval}"
+
+  # Watchdog settings that will take effect on next boot
+  # can be confirmed with mlxbf-bootctl (may be different than
+  # current boot config)
+  if command -v mlxbf-bootctl &> /dev/null; then
+    next_mode=$(mlxbf-bootctl | grep "watchdog mode" | cut -d " " -f 4)
+    next_interval=$(mlxbf-bootctl | grep "watchdog interval" | cut -d " " -f 4)
+  fi
+
+  if [ "${next_mode}" = "disabled" ]; then
```

Comment:
Can you run `checkbashisms` against this script?

Reply:
Done. There were other errors in the file causing `checkbashisms` to exit before reaching the code I added so I just ran it against a modified diff after removing the diff markers. It only came up with one change to use `2>&1` instead of `&>` on line `971`